server/store/datastore/repo.go

```diff
@@ -86,25 +88,42 @@ func (s storage) RepoBatch(repos []*model.Repo) error {
 			log.Debug().Msgf("skip insert/update repo: %#v", repos[i])
 			continue
 		}
+
 		exist, err := sess.
 			Where("repo_owner = ? AND repo_name = ?", repos[i].Owner, repos[i].Name).
 			Exist(new(model.Repo))
 		if err != nil {
 			return err
 		}
+
 		if exist {
 			if _, err := sess.
 				Where("repo_owner = ? AND repo_name = ?", repos[i].Owner, repos[i].Name).
-				AllCols().
+				Cols("repo_scm", "repo_avatar", "repo_link", "repo_private", "repo_clone", "repo_branch").
 				Update(repos[i]); err != nil {
 				return err
 			}
+
+			_, err := sess.
+				Where("repo_owner = ? AND repo_name = ?", repos[i].Owner, repos[i].Name).
+				Get(repos[i])
+			if err != nil {
+				return err
+			}
 		} else {
 			// only Insert on single object ref set auto created ID back to object
 			if _, err := sess.Insert(repos[i]); err != nil {
 				return err
 			}
 		}
+
+		if repos[i].Perm != nil {
```

Comment:
This means: If we do get permissions for the current users for this repo back, we save those permissions, right?

Reply:
yes and that's what syncer now does